### PR TITLE
to_upper_case / to_lower_case added

### PR DESCRIPTION
**Pull request by CrikeeIP**

### CODE REVIEW

File: include/fplus/string_tools.hpp [hunk 1]
```diff
@@ -125,6 +125,28 @@ String trim_whitespace(const String& str)
     return trim_by(is_whitespace<String>, str);
 }
 
+// API search type: to_lower_case : String -> String
+// to_lower_case("ChaRacTer&WorDs23") == "character&words23"
+template <typename String>
+std::string to_lower_case( const std::string& str ) {
+    return fplus::transform( [](char c) -> char{
```
Comment (xtofl) [inline include/fplus/string_tools.hpp]: Was there no way to use `std::tolower`? http://en.cppreference.com/w/cpp/string/byte/toupper

File: include/fplus/string_tools.hpp [hunk 2]
```diff
@@ -125,6 +125,28 @@ String trim_whitespace(const String& str)
...
+            return c -'A' + 'a';
+        }
+    }, s );
+}
+
+// API search type: to_upper_case : String -> String
+// to_upper_case("ChaRacTer&WorDs34") == "CHARACTER&WORDS34"
+template <typename String>
+std::string to_upper_case( const std::string& str ) {
+    return fplus::transform( [](char c) -> char{
```
Comment (xtofl) [inline include/fplus/string_tools.hpp]: Why not use `std::tolower`?  http://en.cppreference.com/w/cpp/string/byte/tolower

File: include/fplus/string_tools.hpp [hunk 3]
```diff
@@ -125,6 +125,28 @@ String trim_whitespace(const String& str)
     return trim_by(is_whitespace<String>, str);
 }
 
+// API search type: to_lower_case : String -> String
+// to_lower_case("ChaRacTer&WorDs23") == "character&words23"
+template <typename String>
```
Comment (xtofl) [inline include/fplus/string_tools.hpp]: Why on earth is this a template if only `std::string` is used?

File: include/fplus/string_tools.hpp [hunk 4]
```diff
@@ -125,6 +125,28 @@ String trim_whitespace(const String& str)
...
+    return fplus::transform( [](char c) -> char{
+        if ( c >= 'A' && c <= 'Z' ) {
+            return c -'A' + 'a';
+        }
+    }, str );
+}
+
+// API search type: to_upper_case : String -> String
+// to_upper_case("ChaRacTer&WorDs34") == "CHARACTER&WORDS34"
+template <typename String>
```
Comment (xtofl) [inline include/fplus/string_tools.hpp]: No `template` needed here.

**Comment by Dobiasd** [comment]
@CrikeeIP Thanks. I just did some [adjustments](https://github.com/Dobiasd/FunctionalPlus/commit/5f1f46c3cb4cb96380086e8458c904a029303595).

**Comment by Dobiasd** [comment]
@xtofl Thank you very much. These kinds of code review are very valuable. I [changed the code accordingly](https://github.com/Dobiasd/FunctionalPlus/commit/a6976041abae790125aa1ae723a8ddbb77b04fb5).

**Comment by xtofl** [comment]
Thanks - I fear I left my kindness at home, today... sorry for any
crudeness.

**Pull request merged**